nevergrad/optimization/optimizerlib.py

```diff
@@ -1203,6 +1203,70 @@ def _internal_tell_candidate(self, candidate: base.Candidate, value: float) -> N
             self._population = []
 
 
+class chaining():
+    """
+    A chaining consists in running algorithm 1 during T1, then algorithm 2 during T2, then algorithm 3 during T3, etc.
+    Each algorithm (except the first one) is fed with the entire archive.
+    """
+    
+    def __init__(
+        self,
+        list_of_algorithms,
+        list_of_budgets,
+    ) -> None:
+        # Either we have the budget for each algorithm, or the last algorithm uses the rest of the budget, so:
+        assert len(list_of_algorithms) == len(list_of_budgets) or len(list_of_algorithms) == len(list_of_budgets) + 1 
+
+        # Let us store the two key parameters:
+        self._list_of_budgets = list_of_budgets
+        self._list_of_algorithms = list_of_algorithms
+
+
+    # From the user point of view, the call below is the creation of the optimizer.
+    def __call__(self, instrumentation: Union[int, Instrumentation], budget: Optional[int] = None, num_workers: int = 1) -> base.Optimizer:
+
+        list_of_algorithms = self._list_of_algorithms
+        list_of_budgets = self._list_of_budgets
+
+        # We create the class we want to use.
+        class chain(base.Optimizer):
```

Comment:
`Chain`
also, creating classes inside a function can lead to reaaaaally weird bugs so it should be avoided :s